_posts/2021-11-11-java-release-notes.md

```diff
@@ -0,0 +1,668 @@
+    }
+```
+We use the keyword `module` with the module name to reference the module. 
+
+The next keyword that we use is `exports`. It tells the module system that we are making our `com.reflectoring.io.app.world` package visible outside of our module. 
+
+There are several other keywords can be used:
+* requires
+* requires transitive
+* exports to
+* uses
+* provides with
+* open
+* opens
+* opens to
+
+Out of these we will show only `requires` declaration. Others can be found in the [docs](https://www.oracle.com/corporate/features/understanding-java-9-modules.html).
+
+#### Defining our Second Module
+After we created and exported the `world` module, we can proceed with creating the `hello` module:
+
+```java
+module hello.module {
+    requires world.module;
+}
+```
+We define dependencies using `requires` keyword. We are referencing our newly created, `hello.module`. Packages that are not exported are, by default, module private and cannot be seen from outside of the module. 
+
+To read more about the Java module system please refer to [the docs](https://openjdk.java.net/jeps/261)
+### Try-with-resources Improvement
+Try-with-resources is a feature that enables us to declare new resources on a `try-catch` block to tell the JVM to release these resources after the code has run. The only condition is that the declared resource implements an `Autoclosable` interface.
+#### Manual Closing of Resource
+We want to read text using `BufferedReader`. `BufferedReader` is a closable resource, so we need to make sure that it is properly closed after use. Before Java 8 we would do it like this:
+
+```java
+BufferedReader br = new BufferedReader(new StringReader("Hello world example!"));
+try {
+    System.out.println(br.readLine());
+} catch (IOException e) {
+    e.printStackTrace();
+} finally {
+    try {
+        br.close();
+    } catch (IOException e) {
+        e.printStackTrace();
+    }
+}
+```
+In `finally` block, we would call `close()`. The `finally` block ensures that the reader is always properly closed.
+
+#### Improvement on Autoclosable
+Java 8 introduced the try-with-resource feature that enables us to declare our resource inside `try` definition. This will ensure that our closable is closed without using `finally`. Let us take a look at some example of using the `BufferedReader` to read string:
+```java
+final BufferedReader br3 = new BufferedReader(new StringReader("Hello world example3!"));
+try(BufferedReader reader = br3){
+    System.out.println(reader.readLine());
+}catch (IOException e){
+    System.out.println("Error happened!");
+}
+```
+Inside `try` definition, we assigned our previously created reader to the new variable. Now we know that our reader will get closed every time.
+
+To read more about the try-with-resources feature please refer to [the oracle page.](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)
+### Diamond Syntax with Inner Anonymous Classes
+#### What Java 9 Fixed
```

Comment:
A section with only one subsection is usually a sign that the text structure should be different. I'll remove the subsectio header